pages/instructorAssessmentInstances/instructorAssessmentInstances.ejs

```diff
@@ -93,10 +100,86 @@
-               data-toggle="modal" data-target="#deleteAllAssessmentInstancesModal">
-              <i class="fa fa-times" aria-hidden="true"></i> Delete all assessment instances
-            </button>
+            <div class="dropdown d-flex flex-row">
+              <button type="button" class="btn btn-light dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+                Action for all instances <span class="caret"></span>
+              </button>
+              <div id="PopoverSetTimeLimitAll"
+                   tabindex="0" data-toggle="popover" data-trigger="manual" data-container="body"
+                   data-html="true" data-placement="auto" title="Set Time Limit"
+                   data-content="
+                                 <p>
+                                   Time limit: <%= time_limit_list %><br/>
+                                   Remaining time: <%= remaining_time_range %>
+                                 </p>
+                                 <form name=&quot;set-time-limit-form&quot; method=&quot;POST&quot;>
+                                   <input type=&quot;hidden&quot; name=&quot;__action&quot; value=&quot;set_time_limit_all&quot;>
+                                   <input type=&quot;hidden&quot; name=&quot;__csrf_token&quot; value=&quot;<%= __csrf_token %>&quot;>
+                                   <select class=&quot;form-control select-time-limit&quot; name=plus_minus>
+                                     <% if (remaining_time_min !== null) {
+                                          if (has_open_instance) { %>
+                                     <option value=+1>Add to exams with time limit</option>
+                                     <option value=-1>Subtract from exams with time limit</option>
+                                       <% } else { %>
+                                     <option value=+1>Add</option>
+                                     <option value=-1>Subtract</option>
+                                     <%   }
+                                        } %>
+                                     <option value=set_total>Set time limit to</option>
+                                     <option value=set_rem>Set remaining time to</option>
+                                     <% if (remaining_time_min !== null) { %>
+                                     <option value=unlimited>Set as unlimited</option>
+                                     <% } %>
+                                   </select>
+                                   <p class=&quot;form-inline&quot;>
+                                     <input class=&quot;form-control time-limit-field&quot; type=&quot;number&quot; name=&quot;time_add&quot; style=&quot;width: 5em&quot; value=&quot;5&quot;>
+                                     <select class=&quot;form-control time-limit-field&quot; name=time_ref><option value=60>minutes</option><option value=1>seconds</option></select>
```

Comment:
Ideally I'd like a third option of `percent of total time` here, so that I could easily update everyone's times while preserving the extended-duration for disability accommodations.

Personally, I'm not sure that it's worth bothering with `seconds` as an option. Or is there a use case where `minutes` wouldn't be sufficient?

Reply:
New option (still some tests pending) lets you do weird things like "Set remaining time to 30% of total time limit"...